src/libstore/build/worker.cc

```diff
@@ -380,7 +380,10 @@ void Worker::waitForInput()
         std::set<int> fds2(j->fds);
         std::vector<unsigned char> buffer(4096);
         for (auto & k : fds2) {
-            if (pollStatus.at(fdToPollStatus.at(k)).revents) {
+            auto fdps1 = get(fdToPollStatus, k);
+            assert(fdps1);
+            assert(*fdps1 < pollStatus.size());
+            if (pollStatus.at(*fdps1).revents) {
```

Comment:
I won’t suggest an alternative variable name for `fdps1` because I have no idea what this code is doing, but if you have one, that would be immensely helpful